### ISSUE TITLE
workspaceRoot invalid on Git Bash and WSL

### ISSUE THREAD

**Issue by jdforsythe** [Microsoft/vscode#24072]
- VSCode Version: 1.11.0
- OS Version: Windows 10

Steps to Reproduce:

1. Set default shell to Git Bash or WSL (CTRL+SHIFT+P - "shell")
2. Run a launch task using `${workspaceRoot}`

Here's an example launch config:

```json
{
  "name": "Debug All (node2)",
  "type": "node2",
  "request": "launch",
  "protocol": "inspector",
  "stopOnEntry": false,
  "args": [
    "--nolazy",
    "${workspaceRoot}/debugall.js"
  ],
  "cwd": "${workspaceRoot}/",
  "preLaunchTask": null,
  "runtimeExecutable": null,
  "env": {
    "NODE_ENV": "development"
  },
  "sourceMaps": false,
  "outDir": null,
  "console": "integratedTerminal",
  "skipFiles": [
    "<node_internals>/**",
    "**/node_modules/**"
  ]
},
```

Running this with the Powershell terminal works just fine. When running in WSL or Git Bash, the `${workspaceRoot}` uses backslashes in the path and fails:

```sh
$ cd c:\dev\projectname/ ; env NODE_ENV=development node --inspect=32931
 --debug-brk --nolazy c:\dev\projectname/debugall.js
bash: cd: c:projectname/: No such file or directory
Debugger listening on port 32931.
Warning: This is an experimental feature and could change at any time.
To start debugging, open the following URL in Chrome:
    chrome-devtools://devtools/remote/serve_file/@60cd6e859b9f557d2312f5bf532f6aec5f284980/inspector.html?experiments=true&v8only=true&ws=localhost:32931/afb2c
73f-e645-4693-91bf-fc4b5a6364f8
Debugger attached.
module.js:471
    throw err;
    ^

Error: Cannot find module 'c:\dev\projectname\devprojectname\debugall.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Timeout.Module.runMain [as _onTimeout] (module.js:604:10)
    at ontimeout (timers.js:365:14)
    at tryOnTimeout (timers.js:237:5)
    at Timer.listOnTimeout (timers.js:207:5)
Waiting for the debugger to disconnect...
```

Note that these commands will work in Git Bash if the backslashes are converted to the proper forward-slashes:

```sh
$ cd c:\dev\projectname
bash: cd: c:devprojectname: No such file or directory
$ cd c:/dev/projectname
[no error]
$ node c:/dev/projectname/debugall.js
[runs without error]
```

Converting the slashes doesn't fix WSL, though. In Git Bash the drives are mounted like `/c/` for `c:\`. In WSL they're mounted at `/mnt/c/`, so it may require changing the drive part of `${workspaceRoot}` depending on the shell you have selected. We could hard-code the workspace root, but that forces everyone to use a *nix based shell and have the project in the same path on their different machines.

**Comment by CherryDT** [comment]
I tried the same thing and I did it using a script which converts the slashes for me and then launches node in WSL, but then of course VSCode doesn't understand the paths any longer. It is unfortunately not possible to specify a `remoteRoot` for launch actions (see my request https://github.com/Microsoft/vscode/issues/11681 a long time ago), so my workaround at the moment is to start the applications in the integrated console (which I have configured to open bash by default) using `node --inspect --debug-brk server.js` or similar, and then configuring a "WSL attach" action which has the `remoteRoot` set to `/mnt/c/.../myProject`. This way I can attach the VSCode debugger to the node instance running in WSL.

**Comment by weinand** [comment]
The title is misleading because the value of `${workspaceRoot}` is perfectly fine and follows the spec. So there is nothing we will "fix".

The problem is that it is being used in a context where backslashes in paths are not work they way they normally work in Windows. 

In git/bash a command like `node lib\app.js` does not work because the backslash is not escaped (or the string is not quoted).

I've fixed that issue in #21635 by always quoting arguments that contain backslashes.

@CherryDT your issue/proposal is covered by this PR https://github.com/Microsoft/vscode-node-debug/pull/156

**Comment by weinand** [comment]
duplicate of #21635
